tests/datasource/fluent/conftest.py

```diff
@@ -409,3 +418,61 @@ def seeded_contexts(
     """Parametrized fixture for seeded File and Cloud DataContexts."""
     context_fixture: FileDataContext | CloudDataContext = request.getfixturevalue(request.param)
     return context_fixture
+
+
+@pytest.fixture
+def pandas_filesystem_datasource(empty_data_context) -> PandasFilesystemDatasource:
+    base_directory_rel_path = pathlib.Path("..", "..", "test_sets", "taxi_yellow_tripdata_samples")
+    base_directory_abs_path = (
+        pathlib.Path(__file__).parent.joinpath(base_directory_rel_path).resolve(strict=True)
+    )
+    pandas_filesystem_datasource = PandasFilesystemDatasource(
+        name="pandas_filesystem_datasource",
+        base_directory=base_directory_abs_path,
+    )
+    pandas_filesystem_datasource._data_context = empty_data_context
+    return pandas_filesystem_datasource
+
+
+@contextmanager
+def _source(
+    validate_batch_spec: Callable[[SqlAlchemyDatasourceBatchSpec], None],
+    dialect: str,
+    connection_string: str = "postgresql+psycopg2://postgres:@localhost/test_ci",
+    data_context: Optional[AbstractDataContext] = None,
+    partitioner_query_response: Optional[List[Dict[str, Any]]] = None,
+    create_temp_table: bool = True,
+) -> Generator[PostgresDatasource, None, None]:
+    partitioner_response = partitioner_query_response or (
+        [
+            {"year": year, "month": month}
+            for year in _DEFAULT_TEST_YEARS
+            for month in _DEFAULT_TEST_MONTHS
+        ]
+    )
+
+    execution_eng_cls = sqlachemy_execution_engine_mock_cls(
+        validate_batch_spec=validate_batch_spec,
+        dialect=dialect,
+        partitioner_query_response=partitioner_response,
+    )
+    original_override = PostgresDatasource.execution_engine_override  # type: ignore[misc]
```

Comment:
Not for this PR but it seems like `execution_engine_override` should not be a `ClassVar`. It would be great if we could instantiate the PostgresDatasource without having to manage state with try/finally.

Reply:
Yeah, seems reasonable. And just in case it got lost in all the noise of this PR: this was just moved from a deeper-nested test file.